### PR TITLE
Feature/support proxy for freshclam

### PR DESCRIPTION
**Pull request by asteel-gsa**
Test Case (with ajilaag/clamav-rest image):
```bash
/etc/clamav # ./test.sh 
Replaced proxy server #=> ensuring sed works with new delimiter when testing
Replaced proxy port #=> ensuring sed works with new delimiter when testing
Replaced proxy user and pass #=> ensuring sed works with new delimiter when testing

/etc/clamav # cat freshclam.conf | grep "HTTP*"
# The HTTPProxyServer may be prefixed with [scheme]:// to specify which kind
#   http://     HTTP Proxy. Default when no scheme or proxy type is specified.
#   https://    HTTPS Proxy. (Added in 7.52.0 for OpenSSL, GnuTLS and NSS)
HTTPProxyServer https://proxy-test.com
HTTPProxyPort 54321
HTTPProxyUsername TESTUSER
HTTPProxyPassword TEST/PASS
#HTTPUserAgent SomeUserAgentIdString
```

Usage:
If `$PROXY_SERVER` is not set in an environment variable, then none of this will run
If  `$PROXY_SERVER` is set then replace the value in `freshclam.conf`
- Since port, username and password are not required, only apply them to `freshclam.conf` if `$PROXY_PORT`, `$PROXY_USERNAME` and `$PROXY_PASSWORD` are set in an environment variable

### CODE REVIEW

**Comment by asteel-gsa** [comment]
Merge PR from Upstream into GSA-TTS managed repo.
Follow on: Create workflow to Build, Scan and Tag

**Comment by asteel-gsa** [comment]
Tested the workflow [here](https://github.com/asteel-gsa/clamav-rest/actions/runs/5080051322) which publishes a package in ghcr.io [here](https://github.com/asteel-gsa/clamav-rest/pkgs/container/clamav-rest%2Fclamav)

**Comment by mogul** [comment]
@asteel-gsa can you make `GSA-TTS/fac-admins` administrators and `GSA-TTS/fac-team` writers on this repo?

File: .github/workflows/build-image.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push Image
+        run: docker push --all-tags ghcr.io/${{ env.GH_REPO }}/${{ env.IMAGE }}
+
+      - name: Upload Artifact
```
Comment (mogul) [inline .github/workflows/build-image.yml]: What's this for? Caching for later runs?

**Review by mogul — CHANGES_REQUESTED** [review]
Looks good, but if the artifact we upload isn't actually in use, let's comment as to why it's there or remove it.

File: .github/workflows/build-image.yml (hunk 1 above)
Comment (asteel-gsa) [reply .github/workflows/build-image.yml]: And reusing the artifact, can remove it if we dont feel like we need it, just noticed our base scan-images.yml creates the artifact.

**Comment by asteel-gsa** [comment]
> @asteel-gsa can you make `GSA-TTS/fac-admins` administrators and `GSA-TTS/fac-team` writers on this repo?

I do not have settings on the repo, so I am unsure if I am able to add them

**Pull request merged**